### ISSUE TITLE
CNN model trained on distributed deep learning lose generalization

### ISSUE THREAD

**Issue by zfan2016** [horovod/horovod#1930]
**Environment:**
 Framework: Keras backend with TensorFlow

For a binary classification task, I trained a CNN model using both a single GPU with keras backend with tensorflow, and multiple GPUs through distributed learning with Horovod and still keras backend with tensorflow.
In the first case, I firstly loaded both validation and training datasets before training.
In the later case, I loaded both validation and training data through data-generator.
For both cases, the training/validation/testing datasets and model architecture are exactly same, and I got similar validation accuracy of ~ 0.89
Then I loaded the best model got from the two cases to evaluate accuracy on my testing dataset loaded directly without using data-generator. the model trained using single GPU has testing accuracy close to validation accuracy, but the model trained through distributed learning with horovod has testing accuracy of 0.5
What are the possible mistakes I made during distributed learning such that the model got through distributed learning does NOT have any generalization ability? I want to use distributed learning to train larger training datasets.

**Comment by zfan2016** [comment]
please ignore this problem. Suddenly I realized that the two testing datasets are not exactly same. Now I confirmed that the testing and validation accuracies are comparable for the model trained through distributed learning. (I don't know how to delete this issue)
